AWS Integration Provider

lbb opened the pull request:
Resolves #71

lbb committed:
Added mock aws

lbb committed:
Added aws annotations

lbb committed:
Added aws as provider

lbb committed:
Added annotation retrive

lbb committed:
Fixed synatax errors

Note to @realfake don't work that late!

Commit:
Merge branch 'develop' into feature/aws-integration-provider

lbb committed:
Added AWSDatacenterSpec#

lbb committed:
Added AWSNodeSpec

lbb committed:
Added AWSCloudProviderIntegration

lbb committed:
Added AWSSpec kubermatic secret init

lbb committed:
Added helper functions

lbb committed:
Added creating nodes

lbb committed:
Add list nodes

lbb committed:
Add delete nodes

lbb committed:
Add PrepareCloudSpec

lbb committed:
Grouping AWS dc config

lbb committed:
Updated vendoring

lbb committed:
Fixed wrong grouping type of AWS kubernetes-secrets

lbb committed:
Fixed unused defaultVolumeType

lbb committed:
Deleted magic strings

lbb committed:
Changed default tag name

lbb committed:
Fix make template 'static'

lbb committed:
Moved api.Node creation to own function

lbb committed:
Moved PrepareCloudSpec sshKeys to annotations

lbb committed:
Moved buffer to outside of loop

lbb committed:
Cleanup on new node creation...

Note to @relfake: I need to git more + learn rewriting history

lbb committed:
Fixed node return type

lbb committed:
Removed block device, should have done that before

lbb committed:
Renamed var 'created' to 'createdNodes' for clearness

lbb committed:
Fixed unchecked error of userdata creation

lbb committed:
Fixed wrong instance type

lbb committed:
Removed unused AWSNodeSpec size field

lbb committed:
Fixed AWSNodeSpec SSHkeys to SSHKeyName

lbb committed:
Changed aws template initiation to not panic if template was not found

scruplelesswizard commented on provider/cloud/aws/aws.go:
```diff
@@ -51,6 +51,15 @@ type aws struct {
+	tplPath := "template/coreos/cloud-config-node.yaml"
+	if parsed, err := template.New("cloud-config-node.yaml").Funcs(ktemplate.FuncMap).ParseFiles(tplPath); err != nil {
+		tpl = parsed
+		glog.Errorln("template not found:", err)
```
The path where the template is being looked for would be helpful

lbb committed:
Added aws vpc and subnet initialization

lbb committed:
Renamed AWS.VPVId to AWS.VPCId

lbb committed:
Fixed subent netmask from /16 to /24

lbb committed:
Made template stateless

scheeles committed:
Use string for AccessKeyID

scheeles committed:
Merge branch 'feature/aws-integration-provider' of github.com:kubermatic/api into feature/aws-integration-provider

scheeles committed:
Instantiate nodes

lbb committed:
Added aws setup manual

lbb committed:
Changed instance tagging

lbb committed:
Merge branch 'feature/aws-integration-provider' of github.com:/kubermatic/api into feature/aws-integration-provider

scheeles committed:
Add AWS to datacenter.yaml

scheeles committed:
Merge branch 'feature/aws-integration-provider' of github.com:kubermatic/api into feature/aws-integration-provider

mrIncompetent committed:
Merge remote-tracking branch 'origin/develop' into feature/aws-integration-provider

# Conflicts:
#	glide.lock
#	glide.yaml

mrIncompetent committed:
Fixed glide yaml & removed aws-sdk version from yaml

lbb committed:
Added private key list endpoint

lbb committed:
Added Region from AvailibilityZone method.

scheeles committed:
Update datacenter.yaml

scheeles committed:
Fix AWS in datacenter.yaml

Commit:
Merge branch 'develop' into feature/aws-integration-provider

scheeles committed:
Align config of Region and AvailableZone with DigitalOcean

scheeles committed:
Rename a to aws and node.AWS.Type

lbb committed:
Fixed wrong placement

lbb committed:
Fixed wrong placement

lbb committed:
Merge branch 'feature/aws-integration-provider' of github.com:/kubermatic/api into feature/aws-integration-provider

lbb committed:
Fixed: Can't slice pointer slice

lbb committed:
Added dynamic datacenter AMI

Usage:
Under the datacenters.yaml add under AWS object a ami text field with
the AMI-id.

lbb committed:
Removed unused aws tag

lbb committed:
Added overwrite-host variable when host lookup doesn't work

lbb committed:
Added ami to datacenter file

lbb committed:
Merge branch 'develop' into feature/aws-integration-provider

mrIncompetent committed:
Merge remote-tracking branch 'origin/develop' into feature/aws-integration-provider

mrIncompetent committed:
Merge remote-tracking branch 'origin/develop' into feature/aws-integration-provider

mrIncompetent committed:
Fixed linting warning

lbb committed:
Merge branch 'develop' into feature/aws-integration-provider

lbb committed:
Removed vendor

scruplelesswizard commented:
Where are we at with this?

lbb committed:
Merge branch 'develop' into feature/aws-integration-provider

lbb committed:
Corrected documentation

lbb committed:
Added aws-west-2 to datacenters.yaml

mrIncompetent committed:
Merge remote-tracking branch 'origin/develop' into feature/aws-integration-provider

lbb committed:
Merge branch 'develop' into feature/aws-integration-provider

lbb committed:
Fixed CreateKeyCert new prototype, new hosts param

lbb committed:
Fixed aws us-west-2 region

lbb committed:
Fixed wrong aws template assignment

scruplelesswizard commented on provider/cloud/aws/manual/manual.md:
```diff
@@ -0,0 +1,45 @@
+    * The ARN contains your id `arn:aws:iam::YOUR_USER_ID:user/YOUR_NAME`
+* Remember your ID
+
+## 2. Create a plocy.
```
typo

lbb committed:
FIX: Move aws template creation into loop

lbb committed:
DEBUGGING print statements in aws node creation

lbb committed:
DEBUGGING print statements in aws node creation

mrIncompetent committed:
Merge remote-tracking branch 'origin/develop' into feature/aws-integration-provider

# Conflicts:
#	glide.lock

Commit:
Added more debugging statements

lbb committed:
Add kuermatic VPC Name label

Commit:
Updated debugging statements

Commit:
Updated debugging statements

lbb committed:
Debug statements to glog

mrIncompetent committed:
Merge remote-tracking branch 'origin/feature/aws-integration-provider' into feature/aws-integration-provider

# Conflicts:
#	provider/cloud/aws/provider.go

mrIncompetent committed:
Fixed glide.lock file

lbb committed:
Added debugging print statements to node handler

lbb committed:
Merge branch 'feature/aws-integration-provider' of https://github.com/kubermatic/api into feature/aws-integration-provider

Commit:
Change to fmt.Println

mrIncompetent committed:
Hacky fix to create aws nodes

lbb committed:
Hacky bugfix datacenters.yaml

lbb committed:
Added missing error statement

lbb committed:
Fixed varname linting

Commit:
Added build notification for Slack

Commit:
Intentionally break build

Commit:
Revert "Intentionally break build"

This reverts commit f72070a40efb82c33314c6731064d17bd21f4b86.

Commit:
Revert "Change to fmt.Println"

This reverts commit dd1ec95ff808fd976a0e4f6e23e05dc87a54cbb2.

Commit:
Revert "Added debugging print statements to node handler"

This reverts commit a874ba88e82acef938fe4438167451aeaed9a0e4.

Commit:
Revert "Debug statements to glog"

This reverts commit 02ddc763a64a31a6a15716a2ef9a0c8f0c2ade15.

Commit:
Revert "Updated debugging statements"

This reverts commit 79119f5fd863ab165a4a1ec3724ee1b0993cae55.

Commit:
Revert "Updated debugging statements"

This reverts commit 9fc063143e1b3480d587fcc771e28ed278d92dd4.

Commit:
Revert "Added more debugging statements"

This reverts commit ce8ab6cd02686cd4eec95a45028ceb2f75b9d8a1.

Commit:
Revert "DEBUGGING print statements in aws node creation"

This reverts commit 8691069059302634353aeb858d2f886338b941f3.

mrIncompetent committed:
Added cloud provider cleanup

Commit:
Change AWS hard-coded key name for debugging

mrIncompetent committed:
Added internetgateway creation & deletion

mrIncompetent committed:
Merge remote-tracking branch 'origin/feature/aws-integration-provider' into feature/aws-integration-provider

mrIncompetent committed:
Fixed linting errors

Commit:
Rename AWS cloud-config file

Commit:
Rename DO cloud-config file

Commit:
Change cloud-config template name

Commit:
Fixed nameing of template file

mrIncompetent committed:
Added routeTable and modified cleanUp

mrIncompetent committed:
Merge remote-tracking branch 'origin/feature/aws-integration-provider' into feature/aws-integration-provider

Commit:
Added debugging statements for nodes

mrIncompetent committed:
Use default (main) route table for route creation

mrIncompetent committed:
Merge remote-tracking branch 'origin/feature/aws-integration-provider' into feature/aws-integration-provider

Commit:
Improve debugging code

Commit:
Fix instance tag matching for nodes

Commit:
Use private IP as instance name

Commit:
Use name for nodes instead of priv. ip

Commit:
Use only lowercase letters for node names

Commit:
Use node name to delete from Kubernetes API server

Commit:
Ignore errors when deleting node from Kubernetes

sttts commented on handler/datacenter.go:
```diff
@@ -54,6 +59,60 @@ func datacentersEndpoint(
+		sess := ec2.New(session.New(config))
+		// Begin describing key pairs.
+		keys, err := sess.DescribeKeyPairs(&ec2.DescribeKeyPairsInput{})
+		println("1")
```
println?

scruplelesswizard replied:
Debugging. That commit needs to be reverted

sttts commented:
lgtm

lbb committed:
Bugfix #114 to big userdata

mrIncompetent committed:
Added logging on aws cleanup

Commit:
Revert "Improve debugging code"

This reverts commit 099b926cfbff8dd27b694bac109c79c526ee0746.

Commit:
Revert "Added debugging statements for nodes"

This reverts commit 5e6bc24a2cd8ac0678240b8411c4fc6277fa95a1.

Commit:
Remove debugging stattements

Commit:
Add glog to AWS provider

mrIncompetent committed:
Remove ssh-key from aws-instance creation

mrIncompetent committed:
fix sshkey unmarshling on aws

mrIncompetent committed:
Added static ssh key for debugging

mrIncompetent committed:
Changed sshKeys to sshKeyFingerprint on cloudSpec for aws

mrIncompetent committed:
Remove unused variables

mrIncompetent committed:
Renamed sshKeyFingerprint to sskKeyName on aws

lbb committed:
Added wait for delting instances at CleanUp

lbb committed:
Merge branch 'feature/aws-integration-provider' of https://github.com/kubermatic/api into feature/aws-integration-provider

lbb committed:
Move aws cleanup to speperate function

lbb committed:
Added aws instance wait filtering

mrIncompetent committed:
Fixed duplicate creation of aws resources

mrIncompetent committed:
Merge remote-tracking branch 'origin/feature/aws-integration-provider' into feature/aws-integration-provider

mrIncompetent committed:
Added securityGroup-Rule & Added missing tags

mrIncompetent committed:
Fixed linting issue: getAcl -> getACL

Commit:
Fix typos in manual.md
